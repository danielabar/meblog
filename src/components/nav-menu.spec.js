/**
 * @jest-environment jsdom
 */

import dotenv from "dotenv"
dotenv.config({ path: `.env.test` })
import fs from "fs"

import React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

import NavMenu from "./nav-menu"

describe("NavMenu", () => {
  it("renders search input when SEARCH_ENABLED env var is true", () => {
    const container = render(<NavMenu />)
    expect(container).toMatchSnapshot()

    expect(screen.queryByTestId("search-input")).toBeInTheDocument()
  })

  it("does not render search input when SEARCH_ENABLED env var is false", () => {
    const envConfig = dotenv.parse(fs.readFileSync(".env.search_disabled"))
    for (const k in envConfig) {
      process.env[k] = envConfig[k]
    }

    render(<NavMenu />)
    expect(screen.queryByTestId("search-input")).not.toBeInTheDocument()
  })
})
