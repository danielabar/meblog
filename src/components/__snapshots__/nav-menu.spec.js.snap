// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu renders search input when SEARCH_ENABLED env var is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        data-testid="nav-menu"
      >
        <ul>
          <li
            class="undefined undefined"
          >
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="undefined undefined"
          >
            <a
              href="/learning"
            >
              Learning
            </a>
          </li>
          <li
            class="undefined undefined"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="undefined undefined"
          >
            <div>
              <svg
                fill="currentColor"
                height="1.7rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.7rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <input
                aria-label="Search"
                data-testid="search-input"
                placeholder="Search, eg: Rails"
                type="text"
              />
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      data-testid="nav-menu"
    >
      <ul>
        <li
          class="undefined undefined"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="undefined undefined"
        >
          <a
            href="/learning"
          >
            Learning
          </a>
        </li>
        <li
          class="undefined undefined"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="undefined undefined"
        >
          <div>
            <svg
              fill="currentColor"
              height="1.7rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.7rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              aria-label="Search"
              data-testid="search-input"
              placeholder="Search, eg: Rails"
              type="text"
            />
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
